airbyte-config/config-persistence/src/main/java/io/airbyte/config/persistence/ConfigRepository.java

```diff
@@ -1184,8 +1184,10 @@ private Map<UUID, AirbyteCatalog> findCatalogByHash(final String catalogHash, fi
 
     final Map<UUID, AirbyteCatalog> result = new HashMap<>();
     for (final Record record : records) {
-      final AirbyteCatalog catalog = Jsons.deserialize(
-          record.get(ACTOR_CATALOG.CATALOG).toString(), AirbyteCatalog.class);
+      // We do not apply the on-the-fly migration here because the only caller is getOrInsertActorCatalog
+      // which is using this to figure out if the catalog has already been inserted. Migrating on the fly
+      // here will cause us to add a duplicate each time we check for existence of a catalog.
```

Comment:
This seems risky, granted this is a `private` method so there is less risk, but I wonder if this code should be moved to the `getOrInsertActorCatalog` method to prevent any future issues from occurring.

Reply:
I am open to inlining this in `getOrInsertActorCatalog`. One extra note is that this behavior (the insert only inserts when it's actually different) is currently tested.